code/osapi/osapi.cpp

```diff
@@ -716,8 +716,7 @@ namespace os
 void os_ignore_events() {
 	SDL_Event event;
 	while (SDL_PollEvent(&event)) {
-		// Add event to buffer
-		buffered_events.push_back(event);
```

Comment:
This might be the right approach, if this is the only place this function is called.

Reply:
From what I can tell, `os_ignore_events()` is only called during the loading screen code path. @z64555 or someone who is familiar with SDL should probably double check this.

The main reason I changed this function instead of elsewhere is that its documentation states "removes all pending events and ignores them," but that wasn't true. https://github.com/scp-fs2open/fs2open.github.com/blob/1930943d2d3c5175fbeec5dc9b4b80b832fdf7ac/code/osapi/osapi.h#L67

Reply:
After rereading my message above, I just want to make clear that I don't mean to imply that you, @JohnAFernandez, aren't the right person to double check this!

Reply:
No problem, that does match the expected behavior of the function, according to comments.  There was talk of allowing a few controls to be queued, like alt-f4, since usually that is not pressed by accident, and it will help the user exit the program sooner.   On reflection, I'm not sure that would be an improvement. Saw

 I was just mentioning that this was the main thing that really could go wrong with this pr the way it is.

Now that I'm thinking about it, this behavior would have to be changed back if anyone, in the extreme off chance of course, ever implemented a load screen mini game or model view, like in other games.  It may be slightly better to dump the events when the load screen game state closes.